commit 5aeaa78c0204df410b3bd33ce3811da9a7cc8c0f (refs/changes/70/2670/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-09-27T16:59:26-07:00 (2 years ago)
    
    PUBAPI-1446 CreateMachine soft affinity handling isn't soft
    Reviewed by: Julien Gilli <julien.gilli@joyent.com>
    Approved by: Julien Gilli <julien.gilli@joyent.com>

diff --git a/lib/triton-affinity.js b/lib/triton-affinity.js
index 43478e9..44c5aa3 100644
--- a/lib/triton-affinity.js
+++ b/lib/triton-affinity.js
@@ -957,9 +957,14 @@ function localityFromRulesInfo(opts) {
     var nearServerUuids = null; // Servers that satisfy all near rules.
     nearRules.forEach(function (nearRule) {
         if (nearRule.vms.length === 0) {
-            throw new VError('cannot satisfy affinity rule "%s", '
-                + 'it does not match any instances',
-                exprFromRule(nearRule));
+            if (isSoft) {
+                nearRule.skip = true;
+                return;
+            } else {
+                throw new VError('cannot satisfy affinity rule "%s", '
+                    + 'it does not match any instances',
+                    exprFromRule(nearRule));
+            }
         }
 
         // Eliminate "far" servers from candidacy.
